apis/supernova/src/handler/data-property-handler.js

```diff
@@ -113,72 +270,172 @@ class DataPropertyHandler {
     dimension.qDef.qFieldDefs = [field];
     dimension.qDef.qFieldLabels = label ? [label] : [''];
     dimension.qDef.qSortCriterias = INITIAL_SORT_CRITERIAS;
-
     dimension.qDef.autoSort = true;
 
     return dimension;
   }
 
+  /**
+   * Adds a field dimension to the handler.
+   * @param {string} field - The field definition for the dimension.
+   * @param {string=} label - The field label for the dimension.
+   * @param {qix.NxDimension=} defaults - Default properties for the dimension.
+   * @returns {Promise<qix.NxDimension=>} The result of addDimension.
+   * @description Creates and adds a field dimension.
+   * @memberof DataPropertyHandler
+   * @example
+   * handler.addFieldDimension('A', 'AA');
+   */
   addFieldDimension(field, label, defaults) {
     const dimension = this.createFieldDimension(field, label, defaults);
     return this.addDimension(dimension);
   }
 
+  /**
+   * @param {FieldDimensions[]} args - Array of field dimension args.
```

Comment:
@Caele Do you think it makes sense to define a typedef `FieldDimensions` as I did, to set here instead of giving it the type of `object[]`

/**
   * @typeof {object} FieldDimensions
   * @property {string} field
   * @property {string=} label
   * @property {qix.NxDimension=} defaults
   */

Reply:
Best would be if we could align the whole file, for example I see that:
```
addFieldDimensions: (args:FieldDimensions[]):Promise<Promise<qix.NxDimension[]>
```
compared to:
```
addAlternativeFieldDimension: (field, label, defaults):Promise<qix.NxDimension=>
```
So we are not consistent. The problem with adding a type for one of them is that the other won't have a type.
And that we can't change the function signature, it has to stay the same.
So I think it best to use object here instead, and same for other similar cases. But I'm not sure.